### PR TITLE
Feature Movie Collection

### LINKED ISSUE #97: [FR]: Movie Collection
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Related app

All - Android

### Describe the problem

Show other episodes of a movie collection in movie detail page

### Describe the solution

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hadi-norouzi**
close #97

### CODE REVIEW

**Comment by hadi-norouzi** [comment]
@moallemi It's completed but I had to disable read from cache movie detail because local data is not updated when I add a new filed to MovieDetailResponse. Do you have any suggestions?

File: feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailState.kt [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import io.filmtime.core.ui.common.UiMessage
+import io.filmtime.data.model.MovieCollection
 import io.filmtime.data.model.Ratings
 import io.filmtime.data.model.StreamInfo
 import io.filmtime.data.model.VideoDetail
 
 data class MovieDetailState(
   val isLoading: Boolean = false,
   val isBookmarked: Boolean = false,
+  val hasCollection: Boolean = false,
```
Comment (moallemi) [inline feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailState.kt:12]: We do not need a redundant `hasCollection` field. Checking nubility with `videoDetail.collectionId ` works for us

File: feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt [hunk 2]
```diff
@@ -51,8 +53,17 @@ class MovieDetailViewModel @Inject constructor(
...
         result.fold(
           onSuccess = { data ->
-            _state.update { state -> state.copy(videoDetail = data, isLoading = false) }
+            _state.update { state ->
+              state.copy(
+                videoDetail = data,
+                isLoading = false,
+                hasCollection = data.collectionId != null,
+              )
+            }
```
Comment (moallemi) [inline feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt:62]: We do not need a redundant `hasCollection` field.

File: feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt [hunk 3]
```diff
@@ -51,8 +53,17 @@ class MovieDetailViewModel @Inject constructor(
...
             loadRatings()
+            data.collectionId?.let { id ->
+              loadCollection(id.toInt())
+            }
```
Comment (moallemi) [inline feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt:66]: 💅💅💅 to make it cleaner we can check `data.collectionId` nubility  inside `loadCollection` function

**Review by moallemi — DISMISSED** [review]
Thanks for preparing this PR so quickly and contributing to the repository!

**Comment by moallemi** [comment]
> @moallemi It's completed but I had to disable read from cache movie detail because local data is not updated when I add a new filed to MovieDetailResponse. Do you have any suggestions?

If we want to keep cache functionality we have to think about storing collections as well. But lets decide later.

**Pull request merged**